src/monty/io.py

```diff
@@ -186,7 +185,7 @@ def reverse_readline(
     # For windows, we also read the whole file.
     if file_size < max_mem or isinstance(m_file, gzip.GzipFile) or os.name == "nt":
         for line in reversed(m_file.readlines()):
-            yield line.rstrip()
+            yield line.rstrip() if isinstance(line, str) else line.rstrip().decode()
```

Comment:
Might need further confirm (and unit test the type consistency across different OS and file format): https://github.com/materialsvirtuallab/monty/blob/1270c7b230b43cdbc4f5f184ac645ece31d49b61/src/monty/io.py#L89